staging/src/k8s.io/apiserver/pkg/storage/etcd3/testserver/test_server.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testserver
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/server/v3/embed"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"google.golang.org/grpc"
+)
+
+// getAvailablePort returns a TCP port that is available for binding.
+func getAvailablePorts(count int) ([]int, error) {
+	ports := []int{}
+	for i := 0; i < count; i++ {
+		l, err := net.Listen("tcp", ":0")
+		if err != nil {
+			return nil, fmt.Errorf("could not bind to a port: %v", err)
+		}
+		// It is possible but unlikely that someone else will bind this port before we get a chance to use it.
+		defer l.Close()
+		ports = append(ports, l.Addr().(*net.TCPAddr).Port)
+	}
+	return ports, nil
+}
+
+// NewTestConfig returns a configuration for an embedded etcd server.
+// The configuration is based on embed.NewConfig(), with the following adjustments:
+// * uses free ports for client and peer listeners
+// * cleans up the data directory on test termination
+// * silences server logs other than errors
+func NewTestConfig(t *testing.T) *embed.Config {
+	cfg := embed.NewConfig()
```

Comment:
wonder if we should use ```
// UnsafeNoFsync disables all uses of fsync.
	// Setting this is unsafe and will cause data loss.
	UnsafeNoFsync bool `json:"unsafe-no-fsync"`
```
that is a performance boost not having to fsync, and since this is a single instance I think it will not be dangerous

Reply:
+1 Tests should be good without fsync.